python-package/xgboost/core.py

```diff
@@ -1298,7 +1308,10 @@ def feature_names(self, feature_names: Optional[FeatureNames]) -> None:
 
         # validate feature name
         feature_names = _validate_feature_info(
-            feature_names, self.num_col(), "feature names"
+            feature_names,
```

Comment:
I think we don't need to support `slice`, it's mostly used by the `cv` function, which is not supported for distributed training. I would like to remove the `slice` function when we can deprecate the `cv` function and provide document for alternatives.

Reply:
Sounds good.